### PR TITLE
WFLY-14061 Replace Java EE references in java doc with corresponding …

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Issue: https://issues.redhat.com/browse/WFLY-14061
Link with: https://github.com/wildfly/wildfly/pull/13808

### CODE REVIEW

**Comment by RanabirChakraborty** [comment]
@asoldano I have missed a few places. So, I did an overall check and found out the places that I have missed.
@bstansberry If you can review it :)

CC @scottmarlow

File: docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc [hunk 1]
```diff
@@ -95,10 +95,10 @@ javax/faces/api/1.2
 ----
 
 Both modules have the same module name: "javax.faces.api" but one in the
-main directory is for JSF 2.0 and the one located in the 1.2 directory
-is for JSF 1.2. If there was only one module available, we could simply
+main directory is for Faces 2.0 and the one located in the 1.2 directory
```
Comment (scottmarlow) [inline docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc:98]: `JSF 2.0` should be changed to `Jakarta Server Faces 2.3` or `Jakarta Server Faces`.  SPEC API reference link with correct name is https://jakarta.ee/specifications/faces/2.3/.

The current module names from my local wildfly master are:
```
tree ./system/layers/base/javax/faces
./system/layers/base/javax/faces
└── api
    └── main
        ├── jboss-jsf-api_2.3_spec-3.0.0.SP04.jar
        └── module.xml
```

@fjuma should we delete the documentation that explains the `JSF 1.2` module as that doesn't appear to be built anymore on WildFly master?

File: docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc [hunk 2]
```diff
@@ -123,9 +123,9 @@ this, we create a `jboss-deployment-structure.xml` file in the EAR
 ----
 
 In the "deployment" section, we add the dependency for the
-`javax.faces.api` for the JSF 1.2 module. We also add the dependency for
-the JSF 1.2 module in the sub-deployment section for the WAR and exclude
-the module for JSF 2.0.
+`javax.faces.api` for the Faces 1.2 module. We also add the dependency for
+the Faces 1.2 module in the sub-deployment section for the WAR and exclude
+the module for Faces 2.0.
```
Comment (scottmarlow) [inline docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc:128]: Please see above comment regarding JSF/FACES 1.2/2.0 which I think will be replaced with `Jakarta Server Faces 2.3`.

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java [hunk 3]
```diff
@@ -60,7 +60,7 @@
     static {
         // These classes need to be loaded in order!  Some can't be loaded if the Jakarta Server Faces version is too old.
 
-        try { // all of the following classes are available from JSF 2.0 and JSF 2.1
+        try { // all of the following classes are available from Faces 2.0 and Faces 2.1
```
Comment (scottmarlow) [inline jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java:63]: @fjuma  do you agree that these references to `JSF/Faces 2.0/2.1` should change to `Jakarta Server Faces 2.3`?

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java [hunk 4]
```diff
@@ -70,7 +70,7 @@
             stringToAnnoMap.put(FacesBehavior.class.getName(), FacesBehavior.class);
             stringToAnnoMap.put(FacesBehaviorRenderer.class.getName(), FacesBehaviorRenderer.class);
 
-            // Put JSF 2.2+ annotations below this line if any new ones are to be scanned.
+            // Put Jakarta Server Faces 2.2+ annotations below this line if any new ones are to be scanned.
```
Comment (scottmarlow) [inline jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java:73]: @fjuma do you agree with changing `2.2` to `2.3`?

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/ForwardingApplication.java [hunk 5]
```diff
@@ -57,7 +57,7 @@
 /**
  * @author pmuir
  *
- * Bring this class back to allow JSF 1.2 to be used with WildFly
+ * Bring this class back to allow Faces 1.2 to be used with WildFly
```
Comment (scottmarlow) [inline jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/ForwardingApplication.java:60]: @fjuma what do you recommend here for the `Faces 1.2` reference?  Updating to `Jakarta Server Faces 3.0` does seem to fit exactly.

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/WeldApplicationLegacy.java [hunk 6]
```diff
@@ -35,7 +35,7 @@
 /**
  * @author pmuir
  *
- * Bring this class back to allow JSF 1.2 to be used with WildFly
+ * Bring this class back to allow Faces 1.2 to be used with WildFly
```
Comment (scottmarlow) [inline jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/WeldApplicationLegacy.java:38]: please see previous comment.

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java [hunk 7]
```diff
@@ -121,7 +121,7 @@ private void addJSFAPI(String jsfVersion, ModuleSpecification moduleSpecificatio
         moduleSpecification.addSystemDependency(jsfAPI);
     }
 
-    // Is Jakarta Server Faces spec greater than 1.1?  If we add JSF 1.1 support we'll need this to keep from calling addJSFInjection()
+    // Is Jakarta Server Faces spec greater than 1.1?  If we add Faces 1.1 support we'll need this to keep from calling addJSFInjection()
```
Comment (scottmarlow) [inline jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java:124]: @fjuma Should these references to 1.1 be deleted or updated?

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java [hunk 8]
```diff
@@ -156,7 +156,7 @@ private void addJSFInjection(String jsfVersion, ModuleSpecification moduleSpecif
             } else {
                 JSFLogger.ROOT_LOGGER.loadingJsf2x();
                 jsfInjectionDependency.addImportFilter(PathFilters.getMetaInfFilter(), true);
-                // Exclude JSF 1.2 faces-config.xml to make extra sure it won't interfere with JSF 2.0 deployments
+                // Exclude Faces 1.2 faces-config.xml to make extra sure it won't interfere with JSF 2.0 deployments
```
Comment (scottmarlow) [inline jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java:159]: I think the original `JSF 1.2` comment was the best we can do.  @fjuma do you agree?

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/logging/JSFLogger.java [hunk 9]
```diff
@@ -111,11 +111,11 @@
     DeploymentUnitProcessingException jsfInjectionFailed(String slotName, @Cause Throwable cause);
 
     @LogMessage(level = DEBUG)
-    @Message(id = 17, value = "JSF 1.2 classes detected. Using org.jboss.as.jsf.injection.weld.legacy.WeldApplicationFactoryLegacy.")
+    @Message(id = 17, value = "Faces 1.2 classes detected. Using org.jboss.as.jsf.injection.weld.legacy.WeldApplicationFactoryLegacy.")
```
Comment (scottmarlow) [inline jsf/subsystem/src/main/java/org/jboss/as/jsf/logging/JSFLogger.java:114]: @fjuma what should this log message be changed to?  `Faces 1.2` could be changed back to `JSF 1.2` if that fits.  Or changed to `Jakarta Server Faces 2.3` if that fits/works.

File: docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc (hunk 1 above)
Comment (RanabirChakraborty) [reply docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc:98]: @scottmarlow okay, I got it :) thanks for the update. Will wait for @fjuma to reply back on this.

**Review by yersan — COMMENTED** [review]
Hello @RanabirChakraborty 

Could you also take a look at [Getting_Started_Guide.adoc](https://github.com/wildfly/wildfly/blob/17c8ce147e58a4617dae195dc17ce1fb7fd225b2/docs/src/main/asciidoc/Getting_Started_Guide.adoc)? I've just realized we still have some specifications pointing out to JSR names. We should remove from that table all the JSR and replace the name with the Jakarta ones.

**Comment by yersan** [comment]
@RanabirChakraborty the JSR-67 is "Jakarta SOAP with Attachments"

**Review by yersan — COMMENTED** [review]
Hi @RanabirChakraborty , I still can see references for the following acronyms on the .adoc files. Look for those:
git grep -E -n -i " EJB | JACC | JPA | JSF | JTA " |  grep ".adoc"

I based the search on the acronyms attached to the WFLY-14061 Jira. I believe most of them should be replaced.

**Comment by RanabirChakraborty** [comment]
@yersan Just to avoid confusions in many places I have added the NOTE section at the end of those .adoc files.
Where it is explained what would be the Jakarta EE specifications names for the Java EE ones. 
Which was earlier discussed with Brian.

Tho' I'll revisit again to check. Thanks !

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java (hunk 4 above)
Comment (fjuma) [reply jsf/injection/src/main/java/org/jboss/as/jsf/injection/AnnotationMap.java:73]: We should leave this as is. See above.

File: jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/ForwardingApplication.java (hunk 5 above)
Comment (fjuma) [reply jsf/injection/src/main/java/org/jboss/as/jsf/injection/weld/legacy/ForwardingApplication.java:60]: I think we should leave this as is. We should probably remove this class at some point as I don't think it is used anymore.

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java (hunk 8 above)
Comment (fjuma) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java:159]: Yes.

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/logging/JSFLogger.java (hunk 9 above)
Comment (fjuma) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/logging/JSFLogger.java:114]: This needs to reference `1.2`. There are references to both `Faces 1.2` and `JSF 1.2` in https://jakarta.ee/specifications/faces/3.0/jakarta-faces-3.0.html#a4629 so whichever of these that you think is best is fine.

File: docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc (hunk 1 above)
Comment (bstansberry) [reply docs/src/main/asciidoc/_developer-guide/Migrate_Seam_2_Booking_EAR_Binaries_-_Step_by_Step.adoc:98]: I am doing this for https://github.com/jamezp/wildfly/tree/docs-test

File: jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java (hunk 7 above)
Comment (bstansberry) [reply jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFDependencyProcessor.java:124]: I am doing this for https://github.com/jamezp/wildfly/tree/docs-test

**Pull request merged**